include/boost/redis/connection.hpp

```diff
@@ -392,9 +408,21 @@ class connection {
 
    /// Calls `boost::redis::basic_connection::async_exec`.
    template <class Response, class CompletionToken>
-   auto async_exec(request const& req, Response& resp, CompletionToken token)
+   auto async_exec(request const& req, Response& resp, CompletionToken&& token)
    {
-      return impl_.async_exec(req, resp, std::move(token));
+      return async_exec(req, any_adapter(resp), std::forward<CompletionToken>(token));
+   }
+
+   /// Calls `boost::redis::basic_connection::async_exec`.
+   template <class CompletionToken>
+   auto async_exec(request const& req, any_adapter adapter, CompletionToken&& token)
```

Comment:
I did not document the adapters and I don't see any immediate need if this overload for users, The main idea of this PR is to improve compilation times. I think we should wait until someone asks for an adapter interface, that will only happen when people want to pass their own data structures in.

Reply:
Wasn't this PR actually motivated by a user asking for this exact functionality? See comments in https://github.com/boostorg/redis/issues/128